implementation.js

```diff
@@ -1,9 +1,88 @@
 'use strict';
 var ES = require('es-abstract/es6');
 var supportsDescriptors = require('define-properties').supportsDescriptors;
+var has = require('has');
+var global = require('global-object');
```

Comment:
an alternative to this module would be https://www.npmjs.com/package/system.global which follows the official language spec proposal.